### ISSUE TITLE
[Rule request] Implicitly unwrapped function parameters

### ISSUE THREAD

**Issue by Cyberbeni** [realm/SwiftLint#3774]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Sometimes I use implicitly unwrapped optionals for function parameters to signal that calling shouldn't be called with nil but in case you do, the error will be handled inside. (In case I know that handling nil outside of the function would be the same in all use cases)
I'd like to request a rule to warn me whenever I use an implicitly unwrapped function parameter and it gets force unwrapped in the code but not when I use it safely.

Non triggering examples:
```swift
func foo(bar: URL!) {
    guard let bar = bar else { return }
    print(bar.path)
}
```
```swift
func foo(bar: URL!) {
    print(bar?.path ?? "null")
}
```

Triggering example:
```swift
func foo(bar: URL!) {
    print(bar.path)
}
```
